### ISSUE TITLE
spatialEnhance in Joint clustering Analysis

### ISSUE THREAD

**Issue by alberto-valdeolivas** [edward130603/BayesSpace#107]
Dear BayesSpace team, 

Thanks for developing such a nice and useful tool. 

I am currently using BayesSpace to run a Joint Clustering analysis following the corresponding [vignette](https://edward130603.github.io/BayesSpace/articles/joint_clustering.html). In addition, I would like to compute the enhanced clustering by running `spatialEnhance`. 

Would you recommend running it on the combined single cell experiment object (with the offset) or subset it per sample? 

Best regards,
Alberto.

**Comment by edward130603** [comment]
Either way is ok, but note that you may run into memory and/or runtime issues running the enhanced clustering on the combined object.

**Comment by alberto-valdeolivas** [comment]
Thanks for your reply. 

Best,
Alberto.

**Comment by oospina** [comment]
(May 22, 2025)
Commenting just in case somebody has been struggling with merging and then splitting the samples before `spatialEnhance` with the devel version.
I generated a combined SCE following the "joint clustering" vignette and ran `spatialCluster`.
Then, I subset the combined SCE to the spots corresponding to the first sample. No problem with `spatialEnhance` here.
Then, I proceeded with the second sample and got an error:

```
Fitting model...
[DEBUG] The number of threads is 8
[DEBUG] Identifying neighbors of subspots...
Error: Mat::elem(): index out of bounds
```
What helped me was to subtract the number of spots in sample 1 from `sce$spot.idx`.
Had to do the same with `sce_tmp$spot.neighbors`:

```
sce$spot.neighbors = sapply(sce$spot.neighbors, function(i){
  tmp = as.integer(unlist(str_split(i, ',')))
  tmp = tmp - 1062 # This value (1062) was the  number of spots in sample 1
  tmp = paste0(tmp, collapse=',')
})
```
Run `spatialEnhance` on the modified SCE, and that solved the issue I was having.

I would suggest that the authors complement the joint clustering vignette with the `spatialEnhance` part. My apologies in advance if this is a duplicate request.
